code/modules/core_implant/cruciform/rituals/priest.dm

```diff
@@ -521,3 +521,45 @@
 	if(index == 1 && target.address == text && target.active)
 		if(target.wearer && target.wearer.stat != DEAD)
 			return target
+
+/datum/ritual/cruciform/priest/buy_disk
+	name = "Order armaments"
+	phrase = "I need suggestions for a phrase"
+	desc = "Allows you to spend a point to unlock an NT disk."
+	success_message = "Your prayers have been heard."
+	fail_message = "Your prayers have not been answered."
+	power = 20
+
+/datum/ritual/cruciform/priest/buy_disk/perform(mob/living/carbon/human/H, obj/item/implant/core_implant/C, targets)
+	if (!GLOB.armaments_points > 0)
+		fail("You have no miracle of this type to spend.",H,C)
+		return FALSE
+
+	var/list/OBJS = get_front(H)
+
+	var/obj/machinery/power/eotp/EOTP = locate(/obj/machinery/power/eotp) in OBJS
+	if(!EOTP)
+		fail("You must be in front of the Eye of the Protector.", H, C)
+		return FALSE
+
+	if(!length(EOTP.disk_types))
+		EOTP.disk_reward_update()
+	var/list/disk_choices = list()
+	var/reward_disk
+
+	for(var/i=1; i<=length(EOTP.disk_types);i++)
+		var/str = initial(EOTP.disk_types[i].disk_name)
+		disk_choices["[str]"] = EOTP.disk_types[i]
+	var/disk_chosen = input(H,"What does youre church require?","Armenents") as null | anything in disk_choices
+	if (!disk_chosen)
+		fail("Pick a reward.",H,C)
+		return FALSE
+	reward_disk = disk_choices[disk_chosen]
+	EOTP.disk_types -= reward_disk
+	var/obj/item/_item = new reward_disk(get_turf(EOTP))
+	EOTP.visible_message(SPAN_NOTICE("The [_item.name] appers out of bluespace near the [EOTP]!"))
+	GLOB.armaments_points--
+	log_and_message_admins("[key_name(H)] has orderd a [_item.name]")
```

Comment:
if admins want the message to be formated differently let me know. may be bit vague